prboom2/src/gl_main.c

```diff
@@ -153,6 +156,23 @@ GLfloat cm2RGB[CR_LIMIT + 1][4] =
   {1.00f ,1.00f, 1.00f, 1.00f}, //CR_LIMIT
 };
 
+static void gld_SetSpriteFuzzBlendFunc(spritefuzzmode_t fuzzmode)
+{
+    switch (fuzzmode)
```

Comment:
What about `glBlendFunc(x, GL_ONE_MINUS_SRC_ALPHA);` to replace the body of this method? The value `x` either being set when the fuzz mode changes or an array that takes the fuzz mode as its index, depending on what is more convenient?

Reply:
At that point this method probably doesn't need to exist then.

Reply:
That's a reasonable implementation option too.

The argument to keep it in a function would be that, if we decide to expand the fuzzing effects in the future, a function is almost certainly going to be required to do that (more than just setting the blend function). In that case, we have less work to do to expand the function to configure other settings, beyond just the blending function.

For example, there's an option to add a blending equation too that make other invisibility effects possible, such as an "inverted ghost" look. That could can be added to the function but can't be added to `glBlendFunc` directly. Of course this is just an example.

From my side I don't have any concern about either approach; whichever is preferred is fine with me.

Reply:
I can't say what is "preferred" - I can just leave suggestions. 😉 However, given that more implementations for the fuzzing emulation in GL mode are rather unlikely to be added, I agree with @kraflab that it would be neat to have the three possible values for the first argument to `glBlendFunc()` arranged in an array and use `fuzzmode` as an index. This would leave us without any additional branching in this implementation.

Reply:
Also fine with me. It doesn't really make sense to split up the arguments to `glBlendFunc` so we will need two indexed arrays. I pushed a version that looks like this, and also shows an example of a new option that slightly "brightens" the sprites during blending to show how both `glBlendFunc` factors can be used.

There are many different effects possible, so at some point we may need to bring the function back, but for now I think this is fine also.